sql/update.go

```diff
@@ -90,20 +93,68 @@ func (p *planner) Update(n *parser.Update) (planNode, error) {
 		vals = append(vals, val)
 	}
 
+	if len(cols) != len(vals) {
+		panic("columns not equal updated values")
+	}
+	// Secondary indexes needing updating.
+	var indexes []structured.IndexDescriptor
+	for _, index := range tableDesc.Indexes {
+		for _, id := range index.ColumnIDs {
+			if _, ok := colIDSet[id]; ok {
+				indexes = append(indexes, index)
+				break
+			}
+		}
+	}
+
 	// Update all the rows.
 	b := client.Batch{}
 	for row.Next() {
 		if err := row.Err(); err != nil {
 			return nil, err
 		}
-		// TODO(vivek): update the secondary indexes too.
-		primaryKey, err := encodeIndexKey(index, colMap, row.Values(), indexKey)
+		rowVals := row.Values()
+		primaryIndexKeySuffix, err := encodeIndexKey(primaryIndex, colIDtoRowIndex, rowVals, nil)
+		if err != nil {
+			return nil, err
+		}
+		primaryIndexKey := bytes.Join([][]byte{primaryIndexKeyPrefix, primaryIndexKeySuffix}, nil)
+		// Compute the current secondary index key:value pairs for this row.
+		secondaryIndexEntries, err := encodeSecondaryIndexes(tableDesc.ID, indexes, colIDtoRowIndex, rowVals, primaryIndexKeySuffix)
+		if err != nil {
+			return nil, err
+		}
+		// Compute the new secondary index key:value pairs for this row.
+		//
+		// Update the row values.
+		for i, col := range cols {
+			rowVals[colIDtoRowIndex[col.ID]] = vals[i]
+		}
+		newSecondaryIndexEntries, err := encodeSecondaryIndexes(tableDesc.ID, indexes, colIDtoRowIndex, rowVals, primaryIndexKeySuffix)
 		if err != nil {
 			return nil, err
 		}
-		// Update the columns.
+		if len(secondaryIndexEntries) != len(newSecondaryIndexEntries) {
```

Comment:
i think this is not a worthwhile check; as long as the `indexes` array passed into `encodeSecondaryIndexes` is unchanged, we're good.